include/task_manager.h

```diff
@@ -68,7 +68,6 @@ namespace detail {
 			task_id tid;
 			const task* tsk_ptr = nullptr;
 			{
-				std::lock_guard lock(m_task_mutex);
```

Comment:
If you get rid of the lock here, you can also get rid of the surrounding compound statement block (which was only introduced for the lock I believe), and then you can get rid of the "forward definition" of `tid` and `tsk_ptr`, which is nice.

Reply:
Good catch! There were also some outdated comments related to locking, removed those as well.